### ISSUE TITLE
Incorrect path and filename used when retrieving CustomPageWebLink metadata

### ISSUE THREAD

**Issue by SCWells72** [forcedotcom/cli#1263]
I noticed this one while doing a bit of testing against DLRS. After successfully deploying the latest DLRS metadata into a scratch org, I tried to then retrieve the org metadata for comparison. Everything came down just fine except for the one `CustomPageWebLink` which was retrieved with a curious file path:

```json
{
  "done": true,
  "fileProperties": [
    {
      "fileName": "/path/to/projects/declarative-lookup-rollup-summaries/dlrs/default/main/default/weblinks/Welcome.weblink-meta.xml",
      "fullName": "Welcome",
      "type": "CustomPageWebLink"
    }
  ],
  "messages": [],
  "status": "SUCCEEDED",
  "success": true
} 
```

Notice that `default` is repeated in the path, and the filename isn't correct. The on-disk location of the original file that was pushed into the org is:

```
/path/to/projects/declarative-lookup-rollup-summaries/dlrs/main/weblinks/Welcome.custompageweblink-meta.xml
```

This is another one like #1262 that is most easily reproduced by removing the local version of that file as if it's for initial retrieval from the org and doing the following:

```
sfdx force:source:retrieve -m CustomPageWebLink:Welcome
=== Retrieved Source
FULL NAME  TYPE               PROJECT PATH
─────────  ─────────────────  ───────────────────────────────────────────────────────────
Welcome    CustomPageWebLink  dlrs\default\main\default\weblinks\Welcome.weblink-meta.xml
```

Again, notice the repeated `default` in the path and the incorrect filename.

Let me know if you need anything else from me on this.

**Comment by mshanemc** [comment]
@SCWells72 can you do `sfdx version --verbose` so we can see your plugin versions?

**Comment by SCWells72** [comment]
Sure. Here you go:

```
$ sfdx version --verbose
 CLI Version :
        sfdx-cli/7.124.0

 Architecture:
        win32-x64

 Node Version :
        node-v14.18.1

 Plugin Version:
        @oclif/plugin-autocomplete 0.3.0 (core)
        @oclif/plugin-commands 1.3.0 (core)
        @oclif/plugin-help 3.2.3 (core)
        @oclif/plugin-not-found 1.2.4 (core)
        @oclif/plugin-plugins 1.10.1 (core)
        @oclif/plugin-update 1.5.0 (core)
        @oclif/plugin-warn-if-update-available 1.7.0 (core)
        @oclif/plugin-which 1.0.3 (core)
        @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
        alias 1.1.22 (core)
        apex 0.3.0 (core)
        auth 1.7.4 (core)
        config 1.2.41 (core)
        custom-metadata 1.0.12 (core)
        data 0.6.3 (core)
        generator 1.2.0 (core)
        limits 1.2.2 (core)
        org 1.8.4 (core)
        salesforce-alm 53.1.1 (core)
        schema 1.0.10 (core)
        sfdx-cli 7.124.0 (core)
        source 1.2.5 (core)
        telemetry 1.2.8 (core)
        templates 52.4.0 (core)
        trust 1.0.8 (core)
        user 1.5.2 (core)

 OS and Version:
        Windows_NT 10.0.22000
```

**Comment by WillieRuemmele** [comment]
Hi @SCWells72 I found this old [issue](https://github.com/forcedotcom/cli/issues/606) regarding `CustomPageWebLink` Using his provided repository I was able to:

- `deploy` the entire project
- delete the `force-app/main/default/weblinks` directory
- `retrieve -m  CustomPageWebLinks` into the correct directory. 

```
 ➜  sfdx force:source:retrieve -m CustomPageWebLink:SalesforceSupport
=== Retrieved Source
FULL NAME          TYPE               PROJECT PATH
─────────────────  ─────────────────  ────────────────────────────────────────────────────────────────────────────
SalesforceSupport  CustomPageWebLink  force-app/main/default/weblinks/SalesforceSupport.custompageweblink-meta.xml
```

with the following version
```
 ➜  sfdx version --verbose                                           
 CLI Version : 
	sfdx-cli/7.124.0

 Architecture: 
	darwin-x64

 Node Version : 
	node-v16.9.0

 Plugin Version: 
	@oclif/plugin-autocomplete 0.3.0 (core)
	@oclif/plugin-command-snapshot 3.1.1 (link) /Users/william.ruemmele/projects/oclif/plugin-command-snapshot
	@oclif/plugin-commands 1.3.0 (core)
	@oclif/plugin-help 3.2.3 (core)
	@oclif/plugin-not-found 1.2.4 (core)
	@oclif/plugin-plugins 1.10.1 (core)
	@oclif/plugin-update 1.5.0 (core)
	@oclif/plugin-warn-if-update-available 1.7.0 (core)
	@oclif/plugin-which 1.0.3 (core)
	@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
	alias 1.1.22 (core)
	apex 0.3.0
	auth 1.7.4 (core)
	command-reference 1.3.4
	config 1.2.41 (core)
	custom-metadata 1.0.12 (core)
	data 0.6.3 (core)
	generator 1.2.1 (core)
	limits 1.2.2 (core)
	org 1.8.4 (core)
	release-management 2.5.2
	└─ trust 1.0.10
	salesforce-alm 53.1.1 (core)
	schema 1.0.10 (core)
	sfdx-cli 7.124.0 (core)
	source 1.2.5 (core)
	telemetry 1.2.8 (core)
	templates 52.4.0 (core)
	user 1.5.2 (core)

 OS and Version: 
	Darwin 19.6.0
````

**Comment by mshanemc** [comment]
I've been digging on this one, and there's a few parts.

## suffix difference  
`Welcome.weblink-meta.xml` => `Welcome.custompageweblink-meta.xml`
This was an error in the original implementation of deploy/retrieve for CustomPageWebLink.  According to the [docs](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/custompageweblink.htm), the suffix should have alway been `.weblink`

For backward compatibility, the new version will retrieve files that are new to you (not in your local source) with the correct prefix (`weblink`) but we didn't want to make that a breaking change, so if you *already* have files that end in .custompageweblink it'll put the changes there.  So there's a few types in the new registry that have a [legacySuffix](https://github.com/forcedotcom/source-deploy-retrieve/blob/8a6aab065ce7032ffd1b46df60040e526f9c6dcb/src/registry/metadataRegistry.json#L476) property to support that 

In your example (I think) you're deleting the local source, which explains why it used the new (correct) suffix.

## File location
If you delete the file locally and then retrieve, we don't know where it was originally.  The basic behavior is `force-app` as the default directory, and new files go into `force-app/main/default` (or in other words, `<defaultPkgDir>/main/default`).   In DLRS project, `dlrs/default` is the default directory so the file is saved into `dlrs/default/main/default` when you have no corresponding local source.

I did test retrieving the metadata **while** I still had a local copy, and it found my copy and matches it there.

I know you mentioned deleting the local source as an easy way to repro--did this difference happen **without** that step, because, if so, we have some windows-path-error somewhere and it's not finding the existing local source and incorrectly creating new source in the default directory.
